src/backend/parser/gram.y

```diff
@@ -5681,19 +5693,33 @@ ExtTableElement:
 			| TableLikeClause				{ $$ = $1; }
 			;
 
-/* column def for ext table - doesn't have room for constraints */
-ExtcolumnDef:	ColId Typename
+/* column def for ext table - allowed to have NOT NULL and DEFAULT constraints */
+ExtcolumnDef:	ColId Typename ExtColQualList
 		{
 			ColumnDef *n = makeNode(ColumnDef);
 			n->colname = $1;
 			n->typeName = $2;
 			n->is_local = true;
 			n->is_not_null = false;
-			n->constraints = NIL;
+			n->raw_default = NULL;
+			n->cooked_default = NULL;
+			SplitColQualList($3, &n->constraints, &n->collClause,
```

Comment:
No, there are no COLLATE clauses for external tables. This was used since this is the default way how ColQualList is handled. In this particular case it can probably be replaced with `n->constraints = $3;`